test/python/circuit/library/test_multipliers.py

```diff
@@ -143,7 +142,7 @@ def test_plugins(self):
         """Test setting the HLS plugins for the modular adder."""
 
         # all gates with the plugins we check, including an expected operation
-        plugins = [("cumulative_h18", "ccircuit-.*"), ("qft_r17", "qft")]
+        plugins = [("cumulative_h18", "ch"), ("qft_r17", "mcphase")]
```

Comment:
When simplifying the code for handling controlled-annotated operations (in particular wrapping a single operation into a circuit so that we only need to consider circuits and not both circuits and operations), I have changed the code on how a `control` is applied to a circuit: instead of first converting the circuit into a gate and applying control to this gate, we create a new quantum circuit with the controlled version of each gate. This makes sure that we do not deteriorate the results in the case of a single iteration (and may actually improve the results in the case of a circuit). Hmm, now that I am writing this, they may be a slight problem when controlling a circuit that has a global phase, I will need to fix this.